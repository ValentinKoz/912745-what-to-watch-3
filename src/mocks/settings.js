const genres = [`All genres`, `Adventure`, `Action`, `Comdey`, `Dramas`];

export {genres};
