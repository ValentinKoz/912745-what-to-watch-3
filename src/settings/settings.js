const MONTHS = [`January`, `February`, `March`, `April`, `May`, `June`, `July`, `August`, `September`, `October`, `November`, `December`];

const Path = {
  LOGIN: `/login`,
  FILMS: `/films`,
  MY_LIST: `/myList`,
  MAIN: `/`,
  REVIEW: `/review`,
  PLAYER: `/player`,
  PROMO: `/promo`,
  COMMENTS: `/comments`,
  FAVORITE: `/favorite`,
};

const Namespace = {DATA: `DATA`, STATE: `STATE`, USER: `USER`};

const DISPLAYED_ITEMS = 8;

const ADD_TO_SHOW = 8;

export {MONTHS, ADD_TO_SHOW, DISPLAYED_ITEMS, Namespace, Path};

export const adaptedObject = (item) => {
  return {
    id: `${item.id}`,
    name: item.name,
    poster: item.poster_image,
    preview: item.preview_image,
    backgroundImg: item.background_image,
    backgroundColor: item.background_color,
    video: item.video_link,
    previewVideo: item.preview_video_link,
    description: item.description,
    rating: item.rating,
    scoresCount: item.scores_count,
    director: item.director,
    starring: item.starring,
    runTime: item.run_time,
    genre: item.genre,
    released: item.released,
    isFavorite: item.is_favorite
  };
};

export const adaptedAuth = (item) => {
  return {
    id: `${item.id}`,
    email: item.email,
    name: item.name,
    avatarUrl: item[`avatar_url`],
  };
};

export const getDate = (propDate) => {
  const date = new Date(propDate);
  const month = MONTHS[date.getMonth()];
  const days = date.getDay();
  const year = date.getFullYear();
  return `${month} ${days}, ${year}`;
};

export const getRatingLevel = (rating) => {
  if (rating >= 0 && rating < 3) {
    return `Bad`;
  } else if (rating >= 3 && rating < 5) {
    return `Normal`;
  } else if (rating >= 5 && rating < 8) {
    return `Good`;
  } else if (rating >= 8 && rating < 10) {
    return `Vary good`;
  } else if (rating === 10) {
    return `Awesome`;
  }
  return ``;
};
