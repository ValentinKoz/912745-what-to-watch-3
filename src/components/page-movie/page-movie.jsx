import React from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import {Link} from "react-router-dom";
import TabList from "./../tab-list/tab-list.jsx";
import MoreLikeThis from "./../more-like-this/more-like-this.jsx";
import {withActiveTab} from "../../hocs/with-active-tab/with-active-tab.js";
import {Operation as Data} from "./../../reducer/data/data.js";
import {AuthorizationStatus} from "../../reducer/user/user.js";
import {Namespace} from "./../../settings/settings.js";
import {Path} from "./../../settings/settings.js";

const TabListWrapped = withActiveTab(TabList);

const PageMovie = React.memo((props) => {
  const {film, onCardClickHandle, films, authorizationStatus, changeFavorite} = props;
  const {genre, released, poster, name, backgroundImg, backgroundColor, id, isFavorite} = film;
  return <><section className="movie-card movie-card--full" style={{background: backgroundColor}}>
    <div className="movie-card__hero">
      <div className="movie-card__bg">
        <img src={backgroundImg} alt={name} />
      </div>

      <h1 className="visually-hidden">WTW</h1>

      <header className="page-header movie-card__head">
        <div className="logo">
          <Link to={Path.MAIN} className="logo__link">
            <span className="logo__letter logo__letter--1">W</span>
            <span className="logo__letter logo__letter--2">T</span>
            <span className="logo__letter logo__letter--3">W</span>
          </Link>
        </div>

        <div className="user-block">
          {authorizationStatus === AuthorizationStatus.AUTH ? (<Link to={Path.MY_LIST}>
            <div className="user-block__avatar">
              <img src="../img/avatar.jpg" alt="User avatar" width="63" height="63" />
            </div>
          </Link>) : (<div className="user-block">
            <Link to={Path.LOGIN} className="user-block__link">Sign in</Link>
          </div>
          )}
        </div>
      </header>

      <div className="movie-card__wrap">
        <div className="movie-card__desc">
          <h2 className="movie-card__title">{name}</h2>
          <p className="movie-card__meta">
            <span className="movie-card__genre">{genre}</span>
            <span className="movie-card__year">{released}</span>
          </p>

          <div className="movie-card__buttons">
            <Link to={`${Path.FILMS}/${id}${Path.PLAYER}`} className="btn btn--play movie-card__button" type="button">
              <svg viewBox="0 0 19 19" width="19" height="19">
                <use xlinkHref="#play-s"></use>
              </svg>
              <span>Play</span>
            </Link>
            {isFavorite ?
              (<button onClick={()=> changeFavorite(id, 0)} className="btn btn--list movie-card__button" type="button">
                <svg viewBox="0 0 18 14" width="18" height="14">
                  <use xlinkHref="#in-list"></use>
                </svg>
                <span>My list</span>
              </button>) : (<button onClick={()=> changeFavorite(id, 1)} className="btn btn--list movie-card__button" type="button">
                <svg viewBox="0 0 19 20" width="19" height="20">
                  <use xlinkHref="#add"></use>
                </svg>
                <span>My list</span>
              </button>)}
            {authorizationStatus === AuthorizationStatus.AUTH && (<Link to={`${Path.FILMS}/:${id}${Path.REVIEW}`} className="btn movie-card__button">Add review</Link>)}
          </div>
        </div>
      </div>
    </div>

    <div className="movie-card__wrap movie-card__translate-top">
      <div className="movie-card__info">
        <div className="movie-card__poster movie-card__poster--big">
          <img src={poster} alt={name} width="218" height="327" />
        </div>
        <TabListWrapped film={film}/>
      </div>
    </div>
  </section>
  <MoreLikeThis films={films} currentGenre={genre} currentFilm={film} onCardClickHandle={onCardClickHandle}/>
  </>;
});
const mapStateToProps = (state) => ({
  films: state[Namespace.DATA].films,
  authorizationStatus: state[Namespace.USER].authorizationStatus,
});

const mapDispatchToProps = (dispatch) => ({
  changeFavorite: (id, status) => dispatch(Data.changeFavorite({id, status})),
});


PageMovie.displayName = `PageMovie`;

PageMovie.propTypes = {
  film: PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    poster: PropTypes.string.isRequired,
    preview: PropTypes.string.isRequired,
    backgroundImg: PropTypes.string.isRequired,
    backgroundColor: PropTypes.string.isRequired,
    video: PropTypes.string.isRequired,
    previewVideo: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    rating: PropTypes.number.isRequired,
    scoresCount: PropTypes.number.isRequired,
    director: PropTypes.string.isRequired,
    starring: PropTypes.arrayOf(PropTypes.string).isRequired,
    runTime: PropTypes.number.isRequired,
    genre: PropTypes.string.isRequired,
    released: PropTypes.number.isRequired,
    isFavorite: PropTypes.bool.isRequired,
  }).isRequired,
  onCardClickHandle: PropTypes.func.isRequired,
  changeFavorite: PropTypes.func.isRequired,
  authorizationStatus: PropTypes.string.isRequired,
  films: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    poster: PropTypes.string.isRequired,
    preview: PropTypes.string.isRequired,
    backgroundImg: PropTypes.string.isRequired,
    backgroundColor: PropTypes.string.isRequired,
    video: PropTypes.string.isRequired,
    previewVideo: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    rating: PropTypes.number.isRequired,
    scoresCount: PropTypes.number.isRequired,
    director: PropTypes.string.isRequired,
    starring: PropTypes.arrayOf(PropTypes.string).isRequired,
    runTime: PropTypes.number.isRequired,
    genre: PropTypes.string.isRequired,
    released: PropTypes.number.isRequired,
    isFavorite: PropTypes.bool.isRequired,
  })),
};

export {PageMovie};
export default connect(mapStateToProps, mapDispatchToProps)(PageMovie);
