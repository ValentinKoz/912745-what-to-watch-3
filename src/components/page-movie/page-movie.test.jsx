import React from "react";
import renderer from "react-test-renderer";
import {PageMovie} from "./page-movie.jsx";
import {BrowserRouter} from "react-router-dom";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import {Namespace} from "./../../settings/settings.js";

const mockStore = configureStore([]);

const displayedItems = 8;

const film = {
  id: `1`,
  name: `Name`,
  poster: `https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Snatch.jpg`,
  preview: `https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/snatch.jpg`,
  backgroundImg: `https://htmlacademy-react-3.appspot.com/wtw/static/film/background/Snatch.jpg`,
  backgroundColor: `#FDFDFC`,
  video: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
  previewVideo: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
  description: `text...`,
  rating: 8.0,
  scoresCount: 43241,
  director: `Guy Ritchie`,
  starring: [`Jason Statham`, `Brad Pitt`, `Benicio Del Toro`],
  runTime: 104,
  genre: `Comedy`,
  released: 2000,
  isFavorite: false
};

it(`PageMovie is renderer correctly`, () => {
  const store = mockStore({
    [Namespace.DATA]: {
      films: [film],
      promo: film,
    },
    [Namespace.STATE]: {
      genre: `All genres`,
      displayedItems,
    },
    [Namespace.USER]: {
      authorizationStatus: `NO_AUTH`
    }
  });
  const tree = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <PageMovie authorizationStatus={`NO_AUTH`} authInfo={{}} location={{pathname: `films/player/1`}} onChangeFavorite={()=>{}} films={[film]} onCardClickHandle={() => {}} showPlayer={false} onShowPlayer={() => {}} />
        </BrowserRouter>
      </Provider>, {
        createNodeMock: () => {
          return {};
        }
      }
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
