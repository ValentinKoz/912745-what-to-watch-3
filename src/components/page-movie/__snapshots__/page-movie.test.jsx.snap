// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageMovie is renderer correctly 1`] = `
Array [
  <section
    className="movie-card movie-card--full"
    style={
      Object {
        "background": "#FDFDFC",
      }
    }
  >
    <div
      className="movie-card__hero"
    >
      <div
        className="movie-card__bg"
      >
        <img
          alt="Name"
          src="https://htmlacademy-react-3.appspot.com/wtw/static/film/background/Snatch.jpg"
        />
      </div>
      <h1
        className="visually-hidden"
      >
        WTW
      </h1>
      <header
        className="page-header movie-card__head"
      >
        <div
          className="logo"
        >
          <a
            className="logo__link"
            href="/"
            onClick={[Function]}
          >
            <span
              className="logo__letter logo__letter--1"
            >
              W
            </span>
            <span
              className="logo__letter logo__letter--2"
            >
              T
            </span>
            <span
              className="logo__letter logo__letter--3"
            >
              W
            </span>
          </a>
        </div>
        <div
          className="user-block"
        >
          <div
            className="user-block"
          >
            <a
              className="user-block__link"
              href="/login"
              onClick={[Function]}
            >
              Sign in
            </a>
          </div>
        </div>
      </header>
      <div
        className="movie-card__wrap"
      >
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            Name
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Comedy
            </span>
            <span
              className="movie-card__year"
            >
              2000
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <a
              className="btn btn--play movie-card__button"
              href="/films/player/1"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </a>
            <button
              className="btn btn--list movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlinkHref="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      className="movie-card__wrap movie-card__translate-top"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster movie-card__poster--big"
        >
          <img
            alt="Name"
            height="327"
            src="https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Snatch.jpg"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <nav
            className="movie-nav movie-card__nav"
          >
            <ul
              className="movie-nav__list"
            >
              <li
                className="movie-nav__item movie-nav__item--active"
              >
                <a
                  className="movie-nav__link"
                  href="#"
                  onClick={[Function]}
                >
                  Overview
                </a>
              </li>
              <li
                className="movie-nav__item "
              >
                <a
                  className="movie-nav__link"
                  href="#"
                  onClick={[Function]}
                >
                  Details
                </a>
              </li>
              <li
                className="movie-nav__item "
              >
                <a
                  className="movie-nav__link"
                  href="#"
                  onClick={[Function]}
                >
                  Reviews
                </a>
              </li>
            </ul>
          </nav>
          <div
            className="movie-rating"
          >
            <div
              className="movie-rating__score"
            >
              8
            </div>
            <p
              className="movie-rating__meta"
            >
              <span
                className="movie-rating__level"
              >
                Vary good
              </span>
              <span
                className="movie-rating__count"
              >
                43241 ratings
              </span>
            </p>
          </div>
          <div
            className="movie-card__text"
          >
            <p>
              text...
            </p>
            <p
              className="movie-card__director"
            >
              <strong>
                Director: Guy Ritchie
              </strong>
            </p>
            <p
              className="movie-card__starring"
            >
              <strong>
                Starring: Jason Statham, Brad Pitt, Benicio Del Toro
              </strong>
            </p>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog catalog--like-this"
    >
      <h2
        className="catalog__title"
      >
        More like this
      </h2>
      <div
        className="catalog__movies-list"
      >
         
      </div>
    </section>
  </div>,
]
`;
